apps/react-workshop/src/ButtonGroup.stories.tsx

```diff
@@ -60,53 +60,60 @@ export const Overflow = () => {
   ));
 
   return (
+    <ButtonGroup
+      orientation='horizontal'
+      overflowButton={(overflowStart) => {
+        return (
+          <DropdownMenu
+            menuItems={(close) => {
+              const length = items.length - overflowStart;
+
+              return Array.from({ length }, (_, _index) => {
+                const index = overflowStart + _index;
+
+                return (
+                  <MenuItem
+                    key={index}
+                    onClick={close}
+                    icon={<SvgPlaceholder />}
+                  >
+                    Item #{index}
+                  </MenuItem>
+                );
+              });
+            }}
+          >
+            <IconButton label='More'>
+              <SvgMore />
+            </IconButton>
+          </DropdownMenu>
+        );
+      }}
+    >
+      {items}
+    </ButtonGroup>
+  );
+};
+Overflow.decorators = [
+  (Story) => (
     <>
       <Text variant='small' as='small' isMuted>
-        Resize the viewport to see overflow behavior.
+        Resize the container to see overflow behavior.
       </Text>
       <div
         style={{
-          maxWidth: 'clamp(300px, 50%, 100%)',
+          width: 'min(30rem, 100%)',
           border: '1px solid hotpink',
           padding: 8,
+          overflow: 'hidden',
+          resize: 'inline',
         }}
       >
-        <ButtonGroup
-          orientation='horizontal'
-          overflowButton={(overflowStart) => {
-            return (
-              <DropdownMenu
-                menuItems={(close) => {
-                  const length = items.length - overflowStart;
-
-                  return Array.from({ length }, (_, _index) => {
-                    const index = overflowStart + _index;
-
-                    return (
-                      <MenuItem
-                        key={index}
-                        onClick={close}
-                        icon={<SvgPlaceholder />}
-                      >
-                        Item #{index}
-                      </MenuItem>
-                    );
-                  });
-                }}
-              >
-                <IconButton label='More'>
-                  <SvgMore />
-                </IconButton>
-              </DropdownMenu>
-            );
-          }}
-        >
-          {items}
-        </ButtonGroup>
+        <Story />
       </div>
     </>
-  );
-};
+  ),
+] satisfies StoryDecorator[];
```

Comment:
nit: Can we not rely on ladle-specific APIs? It makes it harder to migrate to a different tool in the future. You can type the `Story` param as `Story: () => React.ReactNode`